lang/java/compiler/src/main/java/org/apache/avro/compiler/schema/Schemas.java

```diff
@@ -141,9 +140,8 @@ public static String getJavaClassName(final Schema schema) {
             visited.put(schema, schema);
             break;
           case RECORD:
-            Iterator<Schema> reverseSchemas = schema.getFields().stream().map(Field::schema)
-                .collect(Collectors.toCollection(ArrayDeque::new)).descendingIterator();
-            terminate = visitNonTerminal(visitor, schema, dq, () -> reverseSchemas);
+            terminate = visitNonTerminal(visitor, schema, dq, () -> schema.getFields().stream().map(Field::schema)
+                .collect(Collectors.toCollection(ArrayDeque::new)).descendingIterator());
```

Comment:
By inlining the creation of the iterator, we avoid an `AvroException` when using the `IsResolvedSchemaVisitor`.
Reason: you can't call `getFields()` on a schema that didn't have its fields set yet.